h2o-assemblies/steam/Jenkinsfile

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/groovy
+
+@Library('test-shared-library') _
+
+def dockerImage
+def branchOrTag
+def steamImage
+
+pipeline {
+    agent { node { label 'linux&&docker' } }
+
+    parameters {
+        string(name: 'BRANCH_OR_TAG', defaultValue: 'master', description: 'Enter branch or tag you want to scan.')
+    }
+
+    options {
+        ansiColor('xterm')
+        timestamps()
+    }
+
+    stages {
+        stage('0. Init'){
+            steps{
+                script{
+                    checkout(
+                        scm: [$class: 'GitSCM',
+                            branches: [[name: '${BRANCH_OR_TAG}'], [name: 'devops/steam-prisma-jenkins']],
+                            userRemoteConfigs: [[url: 'https://github.com/h2oai/h2o-3.git']]
+                        ]
+                    )
+
+                    sh "git checkout devops/steam-prisma-jenkins"
+                    
+                    dir("h2o-assemblies/steam"){
+                        dockerImage = docker.build("node-java","-f Dockerfile .")
+                    }
+                }
+                
+            }
+        }
+        stage('1. Build jar') {
+            steps {
+                script{
+                    dockerImage.inside(){
+                        sh "git checkout ${BRANCH_OR_TAG}"
+                        sh "./gradlew :h2o-assemblies:steam:shadowJar"
+                        archiveArtifacts artifacts: "h2o-assemblies/steam/build/libs/*.jar"
+                    }
+                }
+            }
+        }
+        stage('2. Scan jar using Prisma'){
+            steps {
+                script{
+                    branchOrTag = "${BRANCH_OR_TAG}".replace('/','-')
+                    steamImage = "h2o-assemblies/steam:${BUILD_NUMBER}-${branchOrTag}"
+
+                    sh """
+                        git checkout devops/steam-prisma-jenkins
+                        ls -la h2o-assemblies/steam/build/libs
+                    """
+                    sh "docker build . -t ${steamImage} -f ./h2o-assemblies/steam/Dockerfile.scanningjars"
+
+                    // scan the image
+                    prismaCloudScanImage ca: '',
+                    cert: '',
+                    dockerAddress: 'unix:///var/run/docker.sock',
+                    image: "${steamImage}",
+                    key: '',
+                    logLevel: 'info',
+                    podmanPath: '',
+                    project: '',
+                    resultsFile: 'prisma-cloud-scan-results.json',
+                    ignoreImageBuildTime:true
+                }
+                
+            }
+        }
+        stage('3. Export results to CSV'){
+            steps{
+                sh "curl -k -u ops:OPS@h2oai https://mr-0xz1:8083/api/v1/scans/download?search=${steamImage} > ${steamImage}.csv"
```

Comment:
Not a good practice harcode creds. please fetch through a Jenkins cred

Reply:
Sure